bundle/config/mutator/resolve_resource_references.go

```diff
@@ -18,6 +21,36 @@ func ResolveResourceReferences() bundle.Mutator {
 
 func (m *resolveResourceReferences) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 	errs, errCtx := errgroup.WithContext(ctx)
+	varPath := dyn.NewPath(dyn.Key("var"))
+	p := dyn.NewPattern(
+		dyn.Key("variables"),
+		dyn.AnyKey(),
+		dyn.Key("lookup"),
+	)
+
+	err := b.Config.Mutate(func(v dyn.Value) (dyn.Value, error) {
+		normalized, _ := convert.Normalize(b.Config, v, convert.IncludeMissingFields)
+
+		// Resolve all variable references in the lookup field.
+		return dyn.MapByPattern(v, p, func(p dyn.Path, v dyn.Value) (dyn.Value, error) {
+			return dynvar.Resolve(v, func(path dyn.Path) (dyn.Value, error) {
+				// Rewrite the shorthand path ${var.foo} into ${variables.foo.value}.
+				if path.HasPrefix(varPath) && len(path) == 2 {
+					path = dyn.NewPath(
+						dyn.Key("variables"),
+						path[1],
+						dyn.Key("value"),
+					)
+				}
+
+				return dyn.GetByPath(normalized, path)
+			})
+		})
+	})
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
Could we do this as a separate mutator to keep it separate?

There is also some duplication between this one and the main variable resolver. Any chance we can combine?